### ISSUE TITLE
Can't get information of an user object

### ISSUE THREAD

**Issue by Elli20** [Adldap2/Adldap2#608]
- Adldap2 Version: 9.0
- LDAP Type: Active Directory 
- PHP Version: 7.1.16 

** Description: **

Hey, i am pretty new at programming, AD and LDAP, i hope, the  Adldap2 Version is right.

I am trying to get data (name;email) of all users from a group (cn=NEWS) in an AD.

My search query looked like this: $search = $provider->search()->users()->get();
I cant use methods like $user->getEmail(); they all return null, even tho, there is an object (i dumped it out). 

Also i can dump a single user with his attribute out :

```php
foreach ($search as $specuser) {
    dd($specuser->cn);
}
```

But i cant use an if-query like this  `if($specuser->cn == $substring2)`. It returns nothing, not even null. But it jumps into the query, i echo-ed 'test'.

My code now looks like this. I've made this because in the NEWS group are all users listed, but just as strings, not as objects with their information. 

```php
$users = $provider->search()->where('cn', '=', 'NEWS')->get();

        foreach ($users as $user) {
            $usersgroup = $user->member;
//getting the user list from all users in the News group

            foreach ($usersgroup as $specificuser) {
//Getting the names (type string) of the users for the if-query

                $position = strpos ( $specificuser , 'CN=');
                $substring = substr ( $specificuser , $position + 3);
                $position2 = strpos ( $substring , ',');
                $substring2 = substr ( $substring , 0 , $position2);

                $search = $provider->search()->users()->get();
//getting user objects

                foreach ($search as $specuser) {
                   if($specuser->cn == $substring2) {

//trying (later) to get the information like email and username 
                       dd($specuser->cn);
                   };
```

Please, i really need help with this. I've tried all, i learned, but nothing worked. I am desperate.

**Comment by stevebauman** [comment]
Hi @Elli20, no worries! I'll help you out and get you up and running with what you need.

It sounds like you're just trying to retrieve members of a certain group and retrieve their information (correct me if I'm wrong)? 

Here's how you'd do this:

```php
// We'll retrieve the group first.
$group = $provider->search()->groups()->find('NEWS');

if ($group) {
    // Retrieve the members of the group.
    $members = $group->getMembers();

    // Go through each member of the group
    foreach ($members as $member) {
        // Now we can retrieve the information of each member:
        $member->getCommonName();
        $member->getEmail();
        $member->getDepartment();
    }
}
```

Does this answer your question?

**Comment by Elli20** [comment]
Thanks, that is working until getFirstName();! I get the error : Call to undefined method Adldap\Models\Contact::getEmail()

**Comment by stevebauman** [comment]
Hi @Elli20, the `getFirstName()` method isn't implemented on `Adldap\Models\Contact` models. This is being fixed (see #447).

For the time being, you can check if the member is a `User` or `Contact` before using the method:

```php
foreach ($members as $member) {
    // Now we can retrieve the information of each member:
    $member->getCommonName();

    if ($member instanceof \Adldap\Models\User) {
        $member->getFirstName();
        $member->getLastName();
        $member->getEmail();
        $member->getDepartment();
    }
}
```

I'm working on a heavy upgrade of all documentation, but this will take me a bit 😅 

You can view all methods available on each returned model by viewing the source:

https://github.com/Adldap2/Adldap2/blob/master/src/Models/User.php

https://github.com/Adldap2/Adldap2/blob/master/src/Models/Contact.php

**Comment by Elli20** [comment]
Thank you so much, now it works! :))
